nilearn/maskers/surface_maps_masker.py

```diff
@@ -429,3 +455,231 @@ def inverse_transform(self, region_signals):
+
+        maps_img = self._reporting_data["maps_img"]
+
+        img = self._reporting_data["images"]
+        if img:
+            img = mean_img(img)
+
+        n_maps = self.maps_img_.shape[1]
+        maps_to_be_displayed = range(n_maps)
+        if isinstance(self.displayed_maps, int):
+            if n_maps < self.displayed_maps:
+                msg = (
+                    "`generate_report()` received "
+                    f"{self.displayed_maps} maps to be displayed. "
+                    f"But masker only has {n_maps} maps. "
+                    f"Setting number of displayed maps to {n_maps}."
+                )
+                warnings.warn(category=UserWarning, message=msg, stacklevel=6)
+                self.displayed_maps = n_maps
+            maps_to_be_displayed = range(self.displayed_maps)
+
+        elif isinstance(self.displayed_maps, (list, np.ndarray)):
+            if max(self.displayed_maps) > n_maps:
+                raise ValueError(
+                    "Report cannot display the following maps "
+                    f"{self.displayed_maps} because "
+                    f"masker only has {n_maps} maps."
+                )
+            maps_to_be_displayed = self.displayed_maps
+
+        self._report_content["number_of_maps"] = n_maps
+        self._report_content["displayed_maps"] = list(maps_to_be_displayed)
+        embeded_images = []
+
+        if img is None:
+            msg = (
+                "SurfaceMapsMasker has not been transformed (via transform() "
+                "method) on any image yet. Plotting only maps for reporting."
+            )
+            warnings.warn(msg, stacklevel=6)
+
+        for roi in maps_to_be_displayed:
+            roi = index_img(maps_img, roi)
+            fig = self._create_figure_for_report(roi=roi, bg_img=img)
+            if self._report_content["engine"] == "plotly":
+                embeded_images.append(fig)
+            elif self._report_content["engine"] == "matplotlib":
+                embeded_images.append(figure_to_png_base64(fig))
+                plt.close()
+
+        return embeded_images
+
+    def _create_figure_for_report(self, roi, bg_img):
+        """Create a figure of maps image, one region at a time.
+
+        If transform() was applied to an image, this image is used as
+        background on which the maps are plotted.
+        """
+        import matplotlib.pyplot as plt
+
+        from nilearn.plotting import plot_surf, view_surf
+
+        threshold = 1e-6
+        if self._report_content["engine"] == "plotly":
+            # squeeze the last dimension
+            for part in roi.data.parts:
+                roi.data.parts[part] = np.squeeze(
+                    roi.data.parts[part], axis=-1
+                )
+            fig = view_surf(
+                surf_map=roi,
+                bg_map=bg_img,
+                bg_on_data=True,
+                threshold=threshold,
+                hemi="both",
+            ).get_iframe()
```

Comment:
pass the cmap here:

the report show a divergent colormap and we most likely will be plotting probabilities that work better with a sequential map like inferno

![image](https://github.com/user-attachments/assets/87c4149e-5bb0-4850-92fd-1cd8fae5dc96)

Reply:
also the iframe should probably not be so wide: that's a CSS issue we can maybe fix in a follow up PR

Reply:
> also the iframe should probably not be so wide: that's a CSS issue we can maybe fix in a follow up PR

Ok, for now, I hard-coded a smaller width, but I think Plotly allows the plots to be responsive to window size. Let's do that in a later PR.

Reply:
Better now

<img width="820" alt="Screenshot 2025-01-24 at 10 00 49 AM" src="https://github.com/user-attachments/assets/eace3dab-0f6d-474c-8f58-af6dd4ddc394" />